internal/venice-common/src/main/java/com/linkedin/venice/offsets/OffsetRecord.java

```diff
@@ -423,12 +421,18 @@ public String toString() {
   public String toSimplifiedString() {
     return "OffsetRecord{" + "localVersionTopicOffset=" + getCheckpointedLocalVtPosition()
         + ", latestProducerProcessingTimeInMs=" + getLatestProducerProcessingTimeInMs() + ", isEndOfPushReceived="
-        + isEndOfPushReceived() + ", upstreamOffset=" + getPartitionUpstreamOffsetString() + ", leaderTopic="
+        + isEndOfPushReceived() + ", upstreamOffset=" + getPartitionUpstreamPositionString() + ", leaderTopic="
         + getLeaderTopic() + '}';
   }
 
-  private String getPartitionUpstreamOffsetString() {
-    return this.partitionState.upstreamOffsetMap.toString();
+  private String getPartitionUpstreamPositionString() {
+    StringBuilder sb = new StringBuilder();
+    for (Map.Entry<String, ByteBuffer> entry: this.partitionState.upstreamRealTimeTopicPubSubPositionMap.entrySet()) {
+      String pubSubBrokerAddress = entry.getKey();
+      PubSubPosition position = getCheckpointedRtPosition(pubSubBrokerAddress);
+      sb.append("[").append(pubSubBrokerAddress).append(": ").append(position).append("] ");
+    }
+    return sb.toString();
```

Comment:
`return this.partitionState.upstreamRealTimeTopicPubSubPositionMap.keySet().stream()
    .map(addr -> "[" + addr + ": " + getCheckpointedRtPosition(addr) + "]")
    .collect(Collectors.joining(" "));`